src/reflect/scala/reflect/internal/settings/MutableSettings.scala

```diff
@@ -66,6 +68,9 @@ abstract class MutableSettings extends AbsSettings {
 
   def isScala212: Boolean
   private[scala] def isScala213: Boolean
+
+  @inline final def isDebug: Boolean     = StatisticsStatics.isDebug     && debug
```

Comment:
@dwijnand Which machinery enables treating BooleanSettings as `boolean`s? (and what is the performance impact of that?)

Reply:
```scala
object MutableSettings {
  import scala.language.implicitConversions
  /** Support the common use case, `if (settings.debug) println("Hello, martin.")` */
  @inline implicit def reflectSettingToBoolean(s: MutableSettings#BooleanSetting): Boolean = s.value
}
```

Reply:
And `value` is a non-final, non-@inline method which then calls a non-final `var`s non-@inline accessor method?
Seems like a lot of indirection to get a true or a false from an instance? Does it make any difference performance-wise if you improve on that access-pattern?